tools/graph_sanity_check.py

```diff
@@ -0,0 +1,288 @@
+        fname_dict[etype] = feat_info[1].split(",")
+    return fname_dict
+
+def gen_node_error_prefix(partition, ntype, fname):
+    """ Create an error prefix for reporting node feature/masks.
+
+        The Prefix will be in the following format nypte-fname-partition.
+        So when errors are reported we can easily sort it and the errors from the
+        same node type and same feature will be grouped.
+
+        Parameters
+        ----------
+        partition: int
+            Partition ID.
+        ntype: str
+            Node type.
+        fname: str
+            Feature name.
+
+        Return
+        ------
+            str: the error prefix
+    """
+    return f"[Node type: {ntype}][Feature Name: {fname}][Part {partition}]:"
+
+def gen_edge_error_prefix(partition, etype, fname):
+    """ Create an error prefix for reporting eode feature/masks.
+
+        The Prefix will be in the following format eypte-fname-partition.
+        So when errors are reported we can easily sort it and the errors from the
+        same eode type and same feature will be grouped.
+
+        Parameters
+        ----------
+        partition: int
+            Partition ID.
+        etype: tuple
+            Edge type.
+        fname: str
+            Feature name.
+
+        Return
+        ------
+            str: the error prefix
+    """
+    return f"[Edge type: {etype}][Feature Name: {fname}][Part {partition}]:"
+
+def verify_masks(mask, error_prefix):
+    """ Validate the data of a mask
+
+        Parameters
+        ----------
+        mask: tensor
+            Mask
+        error_prefix: str
+            Error report prefix
+
+        Return
+        ------
+            str: the error message if the mask is invalid otherwise None
+    """
+    try:
+        # DGL use nonzero_1d to translate a mask into indices of nodes/edges.
+        # so we check the mask using dgl.backend.nonzero_1d
+        mask = F.nonzero_1d(mask)
+    except:
+        return f"ERROR: {error_prefix} The mask is not loadable for loadable. " \
+                 "Ideally, the mask should be a 1D bool tensor."
+    return None
+
+def verify_feats(feat, error_prefix):
+    """ Validate the data of a feat
+
+        Parameters
+        ----------
+        feat: tensor
+            Features
+        error_prefix: str
+            Error report prefix
+
+        Return
+        ------
+            str: the error message if the feature has NaNs
+            str: the warning message if the feature is not normalized
+    """
+    # check whether feat has Nan
+    has_nan = th.isnan(feat).any().item()
+    if has_nan:
+        return f"ERROR: {error_prefix} There are NaN values in the feature, please check.", None
+
+    # check whether feat has inf
+    has_inf = th.isinf(feat).any().item()
+    if has_inf:
+        return f"ERROR: {error_prefix} There are infinite values in the feature, please check.", None
+
+    # check whether the value is normalized between -1 to 1
```

Comment:
Is it necessary to customized the normalized range to an input range?

Reply:
No. But it is recommended.